### PR TITLE
Hardfork for slopeswap

### PR DESCRIPTION
**Pull request by monstrobishi**
Added height selection for #79

### CODE REVIEW

**Comment by monstrobishi** [comment]
Please don't review yet, I'm still making changes

**Comment by monstrobishi** [comment]
Ok review please

File: src/masternodes/poolpairs.h [hunk 1]
```diff
@@ -183,10 +183,10 @@ class CPoolPair : public CPoolPairMessage
         return onReclaim(address, resAmountA, resAmountB);
     }
 
-    Res Swap(CTokenAmount in, PoolPrice const & maxPrice, std::function<Res(CTokenAmount const &)> onTransfer);
+    Res Swap(CTokenAmount in, PoolPrice const & maxPrice, std::function<Res(CTokenAmount const &)> onTransfer, bool postBayfrontGardens = false);
```
Comment (uzyn) [inline src/masternodes/poolpairs.h:186]: should the default be `true` instead? It doesn't matter much, but the default should be what's more likely to be more common, which in this case would be `true`.

**Review by uzyn — COMMENTED** [review]
Thanks @monstrobishi

**Review by uzyn — APPROVED** [review]

File: src/masternodes/poolpairs.cpp [hunk 2]
```diff
@@ -150,17 +150,33 @@ Res CPoolPair::Swap(CTokenAmount in, PoolPrice const & maxPrice, std::function<R
...
+            CAmount stepFrom = std::min(chunk, unswapped);
+            arith_uint256 stepFrom256(stepFrom);
+            arith_uint256 stepTo = poolT * stepFrom256 / poolF;
+            poolF += stepFrom256;
+            poolT -= stepTo;
+            unswapped -= stepFrom;
+            swapped += stepTo;
+        }
+    } else {
+        arith_uint256 unswappedA = arith_uint256(unswapped);
```
Comment (ShengguangXiao) [inline src/masternodes/poolpairs.cpp:160]: I think should put post logic in front, the old logic will always be skipped in the future

**Review by ShengguangXiao — CHANGES_REQUESTED** [review]

File: src/masternodes/poolpairs.cpp (hunk 2 above)
Comment (monstrobishi) [reply src/masternodes/poolpairs.cpp:160]: we can refactor this later, it doesn't change the operation

**Pull request merged**